modules/mobile/app/models/mobile/v0/adapters/service.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'common/models/base'
+require 'date'
+
+module Mobile
+  module V0
+    module Adapters
+      class Service
+        # 411 = Podiatry
+        SERVICE_TYPE_IDS = %w[amputation audiology covid optometry outpatientMentalHealth moveProgram foodAndNutrition
+                              clinicalPharmacyPrimaryCare 411 primaryCare homeSleepTesting socialWork].freeze
+
+        def parse(service_eligibilities)
+          SERVICE_TYPE_IDS.collect do |service|
+            request_facilities = []
+            direct_facilities = []
+
+            service_eligibilities.each do |facility|
+              facility_id = facility.facility_id
+              facility_service = facility.services.select { |h| h[:id] == service }.first
```

Comment:
I believe this is equivalent to `facility_service = facility.services.find { |h| h[:id] == service }`.

Reply:
re-wrote this parser logic last minute and evidently didn't do a great job of cleaning it up. thanks for catching it